ansible_events_ui/api/activation.py

```diff
@@ -55,44 +55,85 @@ async def create_activation(
 async def read_activation(
     activation_id: int, db: AsyncSession = Depends(get_db_session)
 ):
-    query = (
-        sa.select(
-            models.activations.c.id,
-            models.activations.c.name,
-            models.activations.c.description,
-            models.activations.c.activation_enabled,
-            models.activations.c.activation_status,
-            models.activations.c.restarted_at,
-            models.activations.c.restarted_count,
-            models.activations.c.created_at,
-            models.activations.c.modified_at,
-            models.rulebooks.c.id.label("rulebook_id"),
-            models.rulebooks.c.name.label("rulebook_name"),
-            models.inventories.c.id.label("inventory_id"),
-            models.inventories.c.name.label("inventory_name"),
-            models.extra_vars.c.id.label("extra_var_id"),
-            models.extra_vars.c.name.label("extra_var_name"),
-            models.playbooks.c.id.label("playbook_id"),
-            models.playbooks.c.name.label("playbook_name"),
-            models.restart_policies.c.id.label("restart_policy_id"),
-            models.restart_policies.c.name.label("restart_policy_name"),
-            models.execution_envs.c.id.label("execution_env_id"),
-            models.execution_envs.c.url.label("execution_env_url"),
-        )
-        .select_from(
-            models.activations.join(models.rulebooks)
-            .join(models.inventories)
-            .join(models.extra_vars)
-            .join(models.playbooks)
-            .join(models.restart_policies)
-            .join(models.execution_envs)
+    activation = (
```

Comment:
I think the data can still be fetched in a single query. Like this:

```
        sa.select(
            models.activations,
            models.extra_vars,
            models.playbooks,
            models.restart_policies,
            models.execution_envs,
        )
        .select_from(
            models.activations.join(models.rulebooks)
            .join(models.inventories)
            .join(models.extra_vars)
            .join(models.playbooks)
            .join(models.restart_policies)
            .join(models.execution_envs)
        )
        .where(models.activations.c.id == activation_id)
```

Reply:
I have tried the above approach and got the following errors:
```
pydantic.error_wrappers.ValidationError: 8 validation errors for ActivationRead
response -> rulebook
  field required (type=value_error.missing)
response -> inventory -> name
  field required (type=value_error.missing)
response -> inventory -> inventory
  field required (type=value_error.missing)
response -> extra_var -> name
  field required (type=value_error.missing)
response -> extra_var -> extra_var
  field required (type=value_error.missing)
response -> playbook
  value is not a valid dict (type=type_error.dict)
response -> restart_policy
  field required (type=value_error.missing)
response -> execution_env
  field required (type=value_error.missing)

```
I'm not sure if the query above returns response that matches the schema we have. Any thoughts? @cutwater

Reply:
Can you show me the code including serialization logic ?